functests/3_performance_status/status.go

```diff
@@ -133,4 +135,52 @@ var _ = Describe("Status testing of performance profile", func() {
 			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 		})
 	})
+
+	Context("Status reports degraded condition", func() {
+		It("Should report Degraded status if overlapping cpus are configured", func() {
+			if discovery.Enabled() {
+				Skip("Discovery mode enabled, test skipped because it creates incorrect profiles")
+			}
+
+			newRole := "worker-overlapping"
+			newLabel := fmt.Sprintf("%s/%s", testutils.LabelRole, newRole)
+
+			reserved := performancev1.CPUSet("0-3")
+			isolated := performancev1.CPUSet("0-7")
+
+			overlappingProfile := &performancev1.PerformanceProfile{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "PerformanceProfile",
+					APIVersion: performancev1.SchemeGroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "profile-overlapping-cpus",
+				},
+				Spec: performancev1.PerformanceProfileSpec{
+					CPU: &performancev1.CPU{
+						Reserved: &reserved,
+						Isolated: &isolated,
+					},
+					NodeSelector: map[string]string{newLabel: ""},
+					RealTimeKernel: &performancev1.RealTimeKernel{
+						Enabled: pointer.BoolPtr(true),
+					},
+					NUMA: &performancev1.NUMA{
+						TopologyPolicy: pointer.StringPtr("restricted"),
+					},
+				},
+			}
+			Expect(testclient.Client.Create(context.TODO(), overlappingProfile)).ToNot(HaveOccurred())
+			defer func() {
+				err := testclient.Client.Delete(context.TODO(), overlappingProfile)
```

Comment:
You should check that the profile is really deleted & that MCP is not updating, otherwise there's a risk that the profile will leak. Check the profile2 test for inspiration

Reply:
Ack. Do we have a way to recover from a leaked profile?

Reply:
Not sure I understand the question, just make sure that when the test exits the profile is really deleted (as we do with the other profile2 test)

Reply:
Sorry, I was too terse. I got it about making sure the profile should not leak. Followup question: should a profile actually leak (because someone like @fromanirh added a broken test!), do we have a way to recover or this means we polluted the tested cluster and the only remedy is let the cluster admin know so it can be fixed manually?

Reply:
It's the tests responsability not to leave leftovers. In many cases this is achieved by cleaning the namespaces.
In the case of perf profiles though, the only way to clean them is to explicilty delete them. Whatever mechanism you come with, it will require adding the profile to be deleted somewhere (because you also don't want to delete them all), so it's still the dev's responsibility to make sure state don't leak (as with all the other cases).

Reply:
Got it and makes sense. Thanks!